### PR TITLE
Integrate device API

### PR DESCRIPTION
**Pull request by pronebird**
Describe **what** this PR changes. **Why** this is wanted. And, if needed, **how** it does it.

Git checklist:

* [x] Describe the change in **`CHANGELOG.md`** under the `[Unreleased]` header.
* [x] Check that commits follow the [Mullvad coding guidelines](https://github.com/mullvad/coding-guidelines)

- Adds new tunnel settings (v2) model and settings manager.
- Switch tunnel manager to using Device & Accounts API.
- Add migration from old tunnel settings (v1) to v2.
- Obsolete old REST API methods.
- Obsolete keychain classes in favour of using Security frameworks directly.
- Switch to using `throw`-ing functions instead of `Result` to reduce friction associated with handling the `Result` type.
- Obsolete `AccountExpiry`.

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)

### CODE REVIEW

**Review by raksooo — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, 4 of 4 files at r4, 7 of 7 files at r5, 8 of 8 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, 3 of 3 files at r9, 1 of 1 files at r10, 1 of 1 files at r11, 1 of 1 files at r12, 1 of 1 files at r13, 16 of 27 files at r14.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: 40 of 57 files reviewed, all discussions resolved (waiting on @raksooo)

**Review by pinkisemils — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, 4 of 4 files at r4, 7 of 7 files at r5, 8 of 8 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, 3 of 3 files at r9, 1 of 1 files at r10, 1 of 1 files at r11.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: 40 of 57 files reviewed, all discussions resolved (waiting on @raksooo)

**Review by pinkisemils — COMMENTED** [review]
> Reviewed 1 of 1 files at r12, 1 of 1 files at r13, 4 of 27 files at r14.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: 44 of 57 files reviewed, all discussions resolved (waiting on @pinkisemils and @raksooo)

___
*[`ios/MullvadVPN/TunnelManager/SetAccountOperation.swift` line 72 at r14](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603#-N3JWkyYE1HYzzlZL2p-:-N3JWkyZ5kN3v7C5Uany:bfcigxc) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/61e4e3261f46b62d0c4b559bc00395d03a3a23b4/ios/MullvadVPN/TunnelManager/SetAccountOperation.swift#L72)):*
> ```Swift
> 
>     override func main() {
>         if let tunnelSettings = self.state.tunnelSettings {
> ```

It is somewhat unlcear how it follows that `self.state.tunnelSettings` is a good proxy for what action this operation should take. Does this imply that whenever `self.state.tunnelSettings` is not nil, the action will always be `unset`?

**Review by pronebird — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: 44 of 57 files reviewed, all discussions resolved (waiting on @pinkisemils and @raksooo)

___
*[`ios/MullvadVPN/TunnelManager/SetAccountOperation.swift` line 72 at r14](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603#-N3JWkyYE1HYzzlZL2p-:-N3JaYSuBBexWZWcTtPa:b-5oy8ua) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/61e4e3261f46b62d0c4b559bc00395d03a3a23b4/ios/MullvadVPN/TunnelManager/SetAccountOperation.swift#L72)):*
<details><summary><i>Previously, pinkisemils (Emīls Piņķis) wrote…</i></summary><blockquote>

It is somewhat unlcear how it follows that `self.state.tunnelSettings` is a good proxy for what action this operation should take. Does this imply that whenever `self.state.tunnelSettings` is not nil, the action will always be `unset`?
</blockquote></details>

Basically it would always unset current account/device stored in tunnel settings (when set) before, for instance, logging into new account with new device. 

I am looking to refactor that part so it's less of a maze but it will take a little time.

**Review by pinkisemils — COMMENTED** [review]
> Reviewed 7 of 27 files at r14.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: 50 of 58 files reviewed, 1 unresolved discussion (waiting on @pinkisemils, @pronebird, and @raksooo)

___
*[`ios/MullvadVPN/TunnelManager/TunnelManager.swift` line 422 at r14](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603#-N3Jgng68oPcLDyeB3F2:-N3Jgng707CLzO7aJiCj:bjuozk2) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/5fce6ed1cc9081536474a84b6913fed5e7f2b6e4/ios/MullvadVPN/TunnelManager/TunnelManager.swift#L422)):*
> ```Swift
> 
>         operation.completionQueue = .main
>         operation.completionHandler = { completion in
> ```

Is the `completionHandler` not assigned here only because it might be nil?

**Review by pinkisemils — COMMENTED** [review]
> Reviewed 16 of 27 files at r14.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: 50 of 58 files reviewed, 1 unresolved discussion (waiting on @pinkisemils, @pronebird, and @raksooo)

**Review by pinkisemils — APPROVED** [review]
> Reviewed 5 of 5 files at r15, 1 of 1 files at r16, 2 of 2 files at r17, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: all files reviewed, 1 unresolved discussion (waiting on @pronebird)

**Review by pinkisemils — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: all files reviewed, 1 unresolved discussion (waiting on @pronebird)

___
*[`ios/MullvadVPN/TunnelManager/TunnelManager.swift` line 959 at r17](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603#-N3Jy2RfCU0Evcrd7Wkp:-N3Jy2RfCU0Evcrd7Wkq:b-xfqupa) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/1afb6cdf72ad96131fc9a66d585d06adcd3c245b/ios/MullvadVPN/TunnelManager/TunnelManager.swift#L959)):*
> ```Swift
>             where serverErrorResponse?.code == .invalidAccount ||
>             serverErrorResponse?.code == .deviceNotFound:
>             // Do not retry if account or device were removed.
> ```

On desktop, this triggers the GUI to show the revoked device screen.

**Review by pronebird — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: all files reviewed, 1 unresolved discussion (waiting on @pinkisemils)

___
*[`ios/MullvadVPN/TunnelManager/TunnelManager.swift` line 422 at r14](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603#-N3Jgng68oPcLDyeB3F2:-N3JzqnW5AAW1yCSajyH:bmlaee7) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/5fce6ed1cc9081536474a84b6913fed5e7f2b6e4/ios/MullvadVPN/TunnelManager/TunnelManager.swift#L422)):*
<details><summary><i>Previously, pinkisemils (Emīls Piņķis) wrote…</i></summary><blockquote>

Is the `completionHandler` not assigned here only because it might be nil?
</blockquote></details>

`completionHandler` takes as an argument `Error?` but `operation.completionHandler` takes `OperationCompletion` as an argument, so I cannot assign it directly.
___
*[`ios/MullvadVPN/TunnelManager/TunnelManager.swift` line 959 at r17](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603#-N3Jy2RfCU0Evcrd7Wkp:-N3K-BLJBRKE6dKNnRLJ:b-7t4h0t) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/1afb6cdf72ad96131fc9a66d585d06adcd3c245b/ios/MullvadVPN/TunnelManager/TunnelManager.swift#L959)):*
<details><summary><i>Previously, pinkisemils (Emīls Piņķis) wrote…</i></summary><blockquote>

On desktop, this triggers the GUI to show the revoked device screen.
</blockquote></details>

Down the road, I plan to introduce a delegate to tell the UI to reset. However this PR is big enough to complicate even further.

**Review by pinkisemils — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/3603)* status: :shipit: complete! all files reviewed, all discussions resolved
